src/coreclr/jit/ssabuilder.cpp

```diff
@@ -840,28 +839,34 @@ void SsaBuilder::RenameDef(GenTree* defNode, BasicBlock* block)
 
     GenTreeLclVarCommon* lclNode;
     bool                 isFullDef = false;
-<<<<<<< HEAD
+    ssize_t              offset    = 0;
+    unsigned             storeSize = 0;
 #if defined(TARGET_WASM)
     bool isLocal;
     if (block->IsLIR())
     {
-        isLocal   = defNode->OperIsLocalStore();
-        lclNode   = isLocal ? defNode->AsLclVarCommon() : nullptr;
-        isFullDef = isLocal && (lclNode->OperIs(GT_STORE_LCL_VAR) ||
-                        (m_pCompiler->lvaTable[lclNode->GetLclNum()].lvExactSize == genTypeSize(defNode->gtType)));
+        isLocal = defNode->OperIsLocalStore();
+        lclNode = isLocal ? defNode->AsLclVarCommon() : nullptr;
+        isFullDef = !lclNode->IsPartialLclFld(m_pCompiler);
+
+        if (lclNode->OperIs(GT_STORE_LCL_FLD))
+        {
+            offset = lclNode->AsLclFld()->GetLclOffs();
+            storeSize = lclNode->AsLclFld()->GetSize();
+        }
+        else
+        {
+            offset = lclNode->GetLclOffs();
```

Comment:
```suggestion
            offset = 0;
```

Reply:
@yowl Would you like to accept this suggested change?

Reply:
ah, we've got crossed somewhere, I've added another commit to match the suggestion